lib/assessment.js

```diff
@@ -13,14 +19,25 @@ var min = function min() {
 // and returns the sum of that set
 // assume that all the argument are numbers
 var sum = function sum() {
-
+  var total = 0;
+  for (var i = 0; i < arguments.length; i++) {
+    total = total + arguments[i];
+  }
+  return total;
 };
 
 // Question 3
 // Write a function that returns an array of the specified length
 // with all the elements set to defaultValue
+var defaultArray = [];
 var newArray = function newArray(length, defaultValue) {
-
+var i = 0;
+while (length > 0) {
+  defaultArray[i] = defaultValue;
+  length = length - 1;
```

Comment:
We usually don't want to change the values of the parameters we take into the function unless that is their express purpose. Using it as a loop countdown as you did here is a destructive operation, meaning we lose the original state of the parameter and can't reconstruct it afterward without another piece of information.

This can be resolved by changing the loop condition.